src/components/MyPetForm/MyPetForm.jsx

```diff
@@ -1,47 +1,66 @@
 import React, { useState } from 'react';
-import { useNavigate } from "react-router-dom";
+import { useNavigate } from 'react-router-dom';
 
 import { Input, InputMonthAndYear, InputSelect } from '../Inputs';
 import { CustomButton } from '../Button/CustomButton';
+import { Grid } from '@mui/material';
 
-import { Grid} from '@mui/material';
 import { paths } from '../../config/paths';
 import { db, auth } from '../../config/firebase';
+import { collection, addDoc } from 'firebase/firestore';
 
 export const MyPetForm = () => {
+
   const [name, setName] = useState('');
   const [species, setSpecies] = useState('');
   const [breed, setBreed] = useState('');
   const [petAge, setPetAge] = useState(new Date());
 
-const navigate = useNavigate();
+  const navigate = useNavigate();
   const user = auth.currentUser;
+
+  const [enteredNameTouched, setEnteredNameTouched] = useState(false);
+  const enteredNameIsValid = name.trim() !== '' || name.length >= 30;
+  const nameInputIsInvalid = !enteredNameIsValid && enteredNameTouched;
+
+    const handleSubmit = async (e) => {
+      e.preventDefault();
+      setEnteredNameTouched(true);
+  
+      if (!enteredNameIsValid) {
+        return;
+      }
+  
+      setEnteredNameTouched(false);
   
-  const handleSubmit = (e) => {
-    e.preventDefault();
-    db.collection('users').doc(user.uid).collection('pets')
-      .add({
-        name: name,
-        species: species,
-        breed: breed,
-        petAge: petAge,
-      })
-      .then(() => {
-        navigate(paths.myPets, { replace: true });
-      })
-      .catch((error) => {
-        alert(error.message);
-      });
-    setName('');
-    setSpecies('');
-    setBreed('');
-    setPetAge(new Date());
+      await db.collection('pets', user.uid)
+        .add({
+          user_id: user.uid,
+          name: name,
+          species: species,
+          breed: breed,
+          petAge: petAge,
+        })
+        .then(() => {
+          navigate(paths.myPets, { replace: true });
+        })
+        .catch((error) => {
+          alert(error.message);
+        });
   };
 
   return (
     <form onSubmit={handleSubmit}>
       <Grid container direction="column" alignItems="center" gap="1rem">
-        <Input label="Name" type="text" value={name} setValue={setName} />
+        <Input
+          required={true}
```

Comment:
Same as above

For Boolean when you pass required it will be true, if don't it will be false